modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/api/URLHelper.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Anahide Tchertchian
+ */
+package org.nuxeo.apidoc.api;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Helper for URL validity checks.
+ *
+ * @since 20.2.0
+ */
+public class URLHelper {
+
+    public static boolean isValid(String url) {
+        return isValid(url, true);
+    }
+
+    protected static boolean isValid(String url, boolean redirect) {
+        try {
+            HttpURLConnection huc = (HttpURLConnection) new URL(url).openConnection();
+            if (!redirect) {
+                huc.setInstanceFollowRedirects(false);
+            }
+            huc.setRequestMethod("HEAD");
+            int status = huc.getResponseCode();
+            if (HttpURLConnection.HTTP_OK != status) {
+                if (redirect && (status == HttpURLConnection.HTTP_MOVED_TEMP
+                        || status == HttpURLConnection.HTTP_MOVED_PERM || status == HttpURLConnection.HTTP_SEE_OTHER)) {
+                    // retry
+                    String newUrl = huc.getHeaderField("Location");
+                    return isValid(newUrl, true);
```

Comment:
This part should be handled by `HttpURLConnection` class, why do we need it?

Reply:
Because it does not actually follow redirections.
But maybe i complicated things and i should consider it's valid when i get one of the "redirect" statuses, is that what you're saying?

Reply:
I thought `HttpURLConnection` would handle the redirect as it has a flag for it.

Following redirects could be interesting to detect if end server is up, I would keep it.

Reply:
I'm pretty sure i get a 301 as response by default, that's why i looked up the redirection logics, but i'll double check.

Reply:
I saw some discussion on the web about this flag which does nothing. I'm in favor of keeping the redirection logic and add a comment about this weird thing on the JDK class.

Reply:
alright I'll add a comment and clean up logics so that it follows redirects in all cases :+1: